### PR TITLE
bug: trap JWS/JWT errors from being reported as Sentry Errors

### LINKED ISSUE #610: Trap JWT/JWS error exceptions
https://sentry.prod.mozaws.net/operations/autopush-prod/issues/345791/

Some non-critical JWT errors are leaking into Sentry. We should trap those and just reject the submission with appropriate error codes and info.

### PR DESCRIPTION
**Pull request by jrconlin**
Silence Jose errors from handling the Auth header.

Closes #610 

@bbangert , @pjenvey r?

### CODE REVIEW

File: autopush/web/validation.py [hunk 1]
```diff
@@ -320,8 +321,8 @@ def validate_auth(self, d):
 
         try:
             jwt = extract_jwt(token, public_key)
-        except ValueError:
-            raise InvalidRequest("Invalid Authorization Header",
+        except (ValueError, JOSEError) as ex:
+            raise InvalidRequest("Invalid Authorization Header: %s" % str(ex),
```
Comment (bbangert) [inline autopush/web/validation.py]: Do we really want to tell the user the exact jose error message?

Comment (jrconlin) [inline autopush/web/validation.py]: I was kinda debating that. In interests of clarity and developer relationship, it wouldn't hurt. From a more paranoid PoV, it's never a good idea to give more data than absolutely necessary. I'm trying to be less paranoid, but am absolutely fine with not disclosing the actual error.

**Comment by bbangert** [comment]
r+

**Comment by pjenvey** [comment]
r+

**Pull request merged**